src/Eshopworld.Core/IMessenger.cs

```diff
@@ -124,8 +124,9 @@ public interface IDoPubSub : IPublishEvents, IMessageOperations
         /// <param name="subscriptionName">The name of the reliable subscription we're doing for this event type.</param>
         /// <param name="topicName">The name of the topic the subscription is created for</param>
         /// <param name="batchSize">The size of the batch when reading for a topic subscription - used as the pre-fetch parameter of the message receiver</param>
+        /// <param name="deleteOnIdleDurationInMinutes">number of minutes for a subscription to be deleted when idle. If not provided then duration is infinite (TimeSpan.Max)</param>
         /// <exception cref="InvalidOperationException">Thrown when you attempt to setup multiple callbacks against the same <typeparamref name="T"/> parameter.</exception>
-        Task Subscribe<T>(Action<T> callback, string subscriptionName, string? topicName = null, int batchSize = 10) where T : class;
+        Task Subscribe<T>(Action<T> callback, string subscriptionName, string? topicName = null, int batchSize = 10, int? deleteOnIdleDurationInMinutes = null) where T : class;
```

Comment:
Is the Timespan.Max is equivalent to not having the settings applied at all, i.e. is the current, default behaviour the same as the new default one that will be implemented?

Reply:
When you pass null it will be the same behaviour.
Timespan.Max comes from the MS documentation:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.servicebus.fluent.subscription.definition.iwithdeleteonidle.withdeleteonidledurationinminutes?view=azure-dotnet

Reply:
You can see how this is implemented here:
https://github.com/eShopWorld/messaging/pull/23/files#diff-fb8650fb2071a6e5825bc8bc86d1ba94R93